### PR TITLE
Catch invalid options.format values

### PR DESCRIPTION
**Pull request by marijnh**
Before, it'd nebulously crash when trying to dereference `RESERVED_NAMES_BY_FORMAT` instead.

(It'd be nice if the formats were centrally defined, but since they aren't I decided to just add another list.)

This PR contains:
- [ ] bugfix
- [ ] feature
- [ ] refactor
- [ ] documentation
- [x] other

Are tests included?
- [ ] yes (*bugfixes and features will not be merged without tests*)
- [x] no

Breaking Changes?
- [ ] yes (*breaking changes will not be merged unless absolutely necessary*)
- [x] no

### CODE REVIEW

**Comment by marijnh** [comment]
Okay, yeah, see my PR #2814 wrt to the failing CI run.

**Comment by marijnh** [comment]
I can't get the tests to pass either way (a lot of them time out) locally, so I've force-pushed just to see what the CI makes of it.

**Comment by lukastaegert** [comment]
Seems that CI likes it!

File: src/rollup/index.ts [hunk 1]
```diff
@@ -34,7 +34,7 @@ function checkOutputOptions(options: OutputOptions) {
 		});
 	}
 
-	if (!options.format) {
+	if (['amd', 'cjs', 'system', 'es', 'iife', 'umd'].indexOf(options.format) < 0) {
```
Comment (TrySound) [inline src/rollup/index.ts]: You forgot esm

Comment (TrySound) [reply src/rollup/index.ts]: Never mind. I saw Lukas comment

**Pull request merged**